### ISSUE TITLE
Image Load from a memory buffer

### ISSUE THREAD

**Issue by donangel** [simple2d/simple2d#137]
Hi everyone!

Here's my case: I plan to make a simple game using ImGUI. Since I'm very, very new to OpenGL, this library did catch my eye as a good, more understandable way to handle my simple 2D rendering needs.

I'm still researching theoretically (e.g., I didn't clone the code yet), and I cannot find a way to load an image using a memory buffer. Instead, it requires just a filename (which, under iOS/Android, may cause an issue).

Is there a way to modify/extend the lib to supply it with a memory buffer (where a PNG file's contents lie)? I couldn't find the IMG_Load source...

I really appreciate any help you can provide.

**Comment by tlrasor** [comment]
You're correct. It's not currently in simple2d but SDL does have support for it. The code loading images calls out directly to SDL_image library's _IMG_Load_ function, (https://wiki.libsdl.org/SDL_image/IMG_Load). _IMG_Load_ is only a light wrapper around _SDL_RWops_ facilities, (https://wiki.libsdl.org/SDL2/SDL_RWops), and there is a helper even for handling loading from memory, _SDL_RWFromMem_ (https://wiki.libsdl.org/SDL2/SDL_RWFromMem). 

It would be simple but not completely trivial to add this functionality to simple2d. 

For android/etc, SDL has special support for these platforms. I've not experimented with it myself but you can look into how the file handling is done on android here: https://github.com/libsdl-org/SDL/blob/8719a7653565443772b26c9e68025bc3a72d45f1/src/core/android/SDL_android.c#l681
